pkg/insights/prometheus_rules.go

```diff
@@ -0,0 +1,153 @@
+package insights
+
+import (
+	"context"
+
+	"github.com/openshift/insights-operator/pkg/config/configobserver"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringcli "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	rulesName      string = "insights-prometheus-rules"
+	namespaceName  string = "openshift-insights"
+	durationString string = "5m"
+	info           string = "info"
+
+	insightsDisabledAlert                string = "InsightsDisabled"
+	simpleContentAccessNotAvailableAlert string = "SimpleContentAccessNotAvailable"
+	insightsRecommendationActiveAlert    string = "InsightsRecommendationActive"
+)
+
+// PrometheusRulesControllers listens to the configuration observer and
+// creates or removes the Insights Prometheus Rules definitions accordingly
+type PrometheusRulesController struct {
+	configurator   configobserver.Configurator
+	monitoringCS   monitoringcli.Interface
+	promRulesExist bool
+}
+
+func NewPrometheusRulesController(configurator configobserver.Configurator, kubeConfig *rest.Config) PrometheusRulesController {
+	monitoringCS, err := monitoringcli.NewForConfig(kubeConfig)
+	if err != nil {
+		klog.Warningf("Unable create monitoring client: %v", err)
+	}
+	return PrometheusRulesController{
+		configurator: configurator,
+		monitoringCS: monitoringCS,
+	}
+}
+
+// Start starts listening to the configuration observer
+func (p *PrometheusRulesController) Start(ctx context.Context) {
+	configCh, cancel := p.configurator.ConfigChanged()
+	defer cancel()
+
+	p.checkAlertsDisabled(ctx)
+	for {
+		select {
+		case <-configCh:
+			p.checkAlertsDisabled(ctx)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// checkAlertsDisabled reads the actual config and either creates (if they don't exist) or removes (if they do exist)
+// the "insights-prometheus-rules" definition
+func (p *PrometheusRulesController) checkAlertsDisabled(ctx context.Context) {
+	if p.configurator.Config().DisableInsightsAlerts {
+		if p.promRulesExist {
+			err := p.removeInsightsAlerts(ctx)
+			if err != nil {
+				klog.Errorf("Failed to remove Insights Prometheus rules definition: %v ", err)
+			} else {
```

Comment:
nit: I think we can get rid of this else branch, just returning after the klog.Errorf. Then leaving the last statement with no extra tabulation.
e.g.
```
    if err != nil {
        klog.Errorf("Failed to remove Insights Prometheus rules definition: %v ", err)
        return
    }

    klog.Info("Prometheus rules successfully removed")
    p.promRulesExist = false

}
```

Reply:
TBH, I think that this whole `checkAlertsDisabled` has a lot of space to improve. I'm not a big fan of if-else-if-else-if-else and a big fan of early return...

```go
func (p *PrometheusRulesController) checkAlertsDisabled(ctx context.Context) {
	disableInsightsAlerts := p.configurator.Config().DisableInsightsAlerts

	if disableInsightsAlerts && p.promRulesExist {
		if err := p.removeInsightsAlerts(ctx); err != nil {
			klog.Errorf("Failed to remove Insights Prometheus rules definition: %v", err)
			return
		}
		klog.Info("Prometheus rules successfully removed")
		p.promRulesExist = false
		return
	}

	if !disableInsightsAlerts && !p.promRulesExist {
		if err := p.createInsightsAlerts(ctx); err != nil {
			klog.Errorf("Failed to create Insights Prometheus rules definition: %v", err)
			return
		}
		klog.Info("Prometheus rules successfully created")
		p.promRulesExist = true
	}
}
```

☝️ I didn't tested it.

Reply:
Or if you want, you could use `else if` for the second if block and then remove the early returns.

Reply:
Thanks. Updated. Please check